src/views/SignIn.vue

```diff
@@ -0,0 +1,182 @@
+<template>
+    <div v-if="currentUser">
+    <div v-if="showLogInMessage" class="alert alert-primary" role="alert">
+      Successfully logged-in. Welcome <b>{{ currentUser }}</b>, you will be redirected soon...</div>
+      <br><br>
+    </div>
+    <div v-if="!showLogInMessage"><br><br><br><br></div>
+    <div class="container-signin">
+        <div class="row">
+            <h3>Sign In Menu</h3>
+        </div><br>
+        <div class="row justify-content-center">
+            <div class="col-8">
+                <div class="input-group input-group-lg">
+                    <input type="text" class="form-control"
+                    placeholder="Username" v-model="username">
+                </div>
+            </div>
+        </div><br v-if="!showUsernameError">
+        <div v-if="showUsernameError" class="row justify-content-center">
+            <div class="col-8">
+                <div class="error-message">
+                    Username does not exist.
+                </div>
+            </div>
+        </div><br v-if="showUsernameError">
+        <div class="row justify-content-center">
+            <div class="col-8">
+                <div class="input-group input-group-lg">
+                    <input v-if="!showPassword" type="password" class="form-control"
+                    placeholder="Password" v-model="password">
+                    <input v-if="showPassword" type="text" class="form-control"
+                    placeholder="Password" v-model="password">
+                    <button v-if="showPassword" class="btn btn-outline-primary"
+                    @click.prevent="changePasswordVisibility"><i class="far fa-eye"></i>
+                    </button>
+                    <button v-if="!showPassword" class="btn btn-outline-primary"
+                    @click.prevent="changePasswordVisibility"><i class="far fa-eye-slash"></i>
+                    </button>
+                </div>
+            </div>
+        </div><br v-if="!showPasswordError && dontShowEmptyInputError">
+        <div v-if="showPasswordError" class="row justify-content-center">
+            <div class="col-8">
+                <div class="error-message">
+                    Invalid password.
```

Comment:
Error messages may come from an array, no need to alter html for each error message. Manage array in methods instead of this boolean heavy logic that only help with showing/hiding stuff